CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -804,9 +810,198 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
+                else
+                {
+                    for (int bet = 0; bet < playerBets.Count(); bet++)
+                    {
+                        Pots[0].Total += playerBets[bet];
+                    }
+                }
+            }
+            /// <summary>
+            /// Creates side pots as needed at the end of a round of betting.
+            /// </summary>
+            /// <param name="_currentBet">The amount to add to the pot.</param>
+            /// <param name="playerBets">The list of bets from players still active after the round of betting.</param>
+            /// <param name="playersActive">The list of players eligible to win the main pot, all in players will be subtracted for side pot</param>
+            public void CreateSidePots(int _currentBet, List<int> playerBets, List<int> playersActive)
+            {
+                int allInBet = 0; //initiating variable that will hold all in wager
+                int numBets = playerBets.Count();
+
+                for (int bets = 0; bets < numBets; bets++)
+                {
+                    //at least 1 side pot is needed if function is called, lowest bet will be an all-in bet.
+                    if (bets == 0)
+                    {
+                        //all-in bet added to the pot, other players matched bets added to pot as well
+                        allInBet = playerBets.Min();
+                        AddToPot(allInBet, Enumerable.Repeat(allInBet, playerBets.Count()).ToList(), playersActive);
+
+                        //subtracting all-in bet value from all other bets, the all-in bet value has already been added to a pot
+                        for (int player = 0; player < playerBets.Count(); player++)
+                        {
+                            playerBets[player] -= allInBet; // Subtract the amount put into pot from each bet
+                        }
+                        _currentBet -= allInBet;
+
+                        //modify list of players that are eligible for the new main pot by removing player(s) that are all-in
+                        List<int> allInPlayers = playerBets.Select((value, index) => new { value, index })
+                                   .Where(x => x.value == 0)
+                                   .Select(x => x.index)
+                                   .ToList();
+                        playersActive.RemoveAll(value => allInPlayers.Contains(value));
+                        playerBets.RemoveAll(value => value == 0);
+
+                        //New side pot created. Pot that all-in player can win is shifted to inactive position, no more bets can be added to this pot
+                        Pots.Add(new Pot(PotType.SIDE, Pots[0].Total, Pots[0].EligiblePlayers));
+                        Pots[0].EligiblePlayers = playersActive;
+                        Pots[0].Total = 0; //reseting active pot value to empty
+                    }
+                    else if (playerBets.Count() == 1)
```

Comment:
this clause will always be hit, and will break the loop meaning the rest of the code will never be hit. Did you mean to put a `continue` in this if statement?

Reply:
In the refactored code it should still be that else if statement, it is meant to terminate the method once encountered. It would be encountered in the scenario where only one person has funds remaining to bet in the round.